### ISSUE TITLE
MA011 does not report precise location

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#904]
### Version of the Meziantou.Analyzer NuGet package

2.0.238

### Rule Identifier

MA0011

### Target Framework

.NET 9.0

### C# Language version

4.14.0-3.25412.6 (c167adef)

### Description

MA0011 always reports a location that corresponds with the start of the "line" up to the closing brace of the invocation for which the diagnostic is reported.

It would be better if the start position would be start of the invocation for which the diagnostic is reported.

This is of course a small issue.

### Reproduction Steps

Minimal code:

```c#
using System;
using System.Text;

internal sealed class Program
{
    private static void Main()
    {
        var sb = new StringBuilder();

        sb.Append($"{5:x}")
          .Append($"{DateTime.Now:D}")   // MA0011 is reported for position (10,9,11,39) instead of (11,11,11,39)
          .Append($"{DateTime.Now:ZZ}"); // MA0011 is reported for position (10,9,12,39) instead of (12,11,12,40)
    }
}
```

**Comment by meziantou** [comment]
The rule is reported on the `InvocationExpressionSyntax` which include the expression and argument list. In your case, you have multiple _nested_ invocation that all start at the same location, but have different length.

- The first diagnostic corresponds to the span `sb.Append($"{5:x}").Append($"{DateTime.Now:D}")`
- The second diagnostic corresponds to the span `sb.Append($"{5:x}").Append($"{DateTime.Now:D}").Append($"{DateTime.Now:ZZ}")`

**Comment by drieseng** [comment]
I'm ok to close this one if there's nothing actionable.

**Comment by meziantou** [comment]
Technically, it's actionable (Just add `DiagnosticMethodReportOptions.MethodName`). However, I don't get the value of changing it. Can you explain the reasoning behind this?

**Comment by drieseng** [comment]
It would've simplified (a little) our suppressor for **MA0011**, but since we don't need that approach anymore it's not a big deal.
Feel free to close this issue.

**Comment by meziantou** [comment]
I don't understand how it would simplify the suppressor. You can get the node by span (`FindNode(context.Span, getInnermostNodeForTie: true)`) and find the invocation (check ancestor if needed). I won't be different if the span was on the name or argument list. Once you have the InvocationExpressionSyntax, you can do whatever you want.

**Comment by drieseng** [comment]
A colleague has looked into that part, but thanks for the info.
As I said, thanks to your effort we no longer need the suppressor.
